### PR TITLE
[FE-162] Detect IE CSS Issues

### PR DESCRIPTION
**Pull request by thomas-huston-zocdoc**
@ZocDoc/polaris-devs @Pam-Hermanto-ZocDoc @ashley-casey-zocdoc 

Changes
--
- Updated Bundler to validate the final minified CSS before saving it, checking that the file will not be too big or have too many rules for IE to handle.

JIRA Tickets
--
- [**FE-162** Detect in Bundler when Bundled Styles are possibly too large to support in IE](https://zocdoc.atlassian.net/browse/FE-162)

### CODE REVIEW

File: src/css-validator/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    unqualifiedAttributeSelectors: false,
+    userSpecifiedSelectors: false,
+    importantKeywords: false,
+    floatProperties: false,
+    mediaQueries: false
+};
+
+var whitelistedBundleNames = ['mobile', 'csr', 'pulse'];
+
+var shouldValidate = function(bundle) {
```
Comment (thomas-huston-zocdoc) [inline src/css-validator/index.js]: We don't bother validating if it's a mobile or CSR/Pulse bundle since we don't have to worry about IE limits.

Comment (thomas-huston-zocdoc) [inline src/css-validator/index.js]: In the future we can add an option to explicitly opt a bundle out of this check if we find other things that don't match that naming convention should be excluded, but this is good enough for now.

File: src/css-validator/index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    return true;
+
+};
+
+var validateFileSize = function(bundle, result) {
+
+    var kbSize = result.size / 1024;
+
+    if (kbSize > MAX_FILE_SIZE_KB) {
```
Comment (thomas-huston-zocdoc) [inline src/css-validator/index.js]: This makes sure the file isn't larger than 278kb, after which IE arbitrarily just stops reading the file.

File: src/css-validator/index.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        throw new FileTooBigError(bundle, kbSize);
+    }
+
+};
+
+var validateRules = function(bundle, result) {
+
+    if (result.rules > MAX_CSS_RULES) {
```
Comment (thomas-huston-zocdoc) [inline src/css-validator/index.js]: This make sure there are no more than 4095 rules.

File: src/css-validator/index.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
+var _ = require('underscore');
+var path = require('path');
+var StyleStats = require('stylestats');
```
Comment (Laura-Johannet-ZocDoc) [inline src/css-validator/index.js]: wahoo for leverging node modules!  :wine_glass:

**Pull request merged**